go/mysql/client.go

```diff
@@ -170,6 +170,14 @@ func Connect(ctx context.Context, params *ConnParams) (*Conn, error) {
 			return nil, cr.err
 		}
 	}
+
+	if params.IdleTimeoutMs != 0 {
+		_, err := c.ExecuteFetch(fmt.Sprintf("SET SESSION wait_timeout = %d", params.IdleTimeoutMs / 1000), 0, false)
```

Comment:
We should set this to a slightly higher value so that we're the ones to close the connection. Not sure about how to determine. Maybe 10%? Or a flag 30s?

Reply:
I'll add the lesser of 10% and 30s, that way we're not drastically increasing it.  Other than that, I'll fix up the tests and ping you when it's all wrapped up